src/program/top/top.lua

```diff
@@ -9,23 +9,159 @@ local shm = require("core.shm")
 local counter = require("core.counter")
 local S = require("syscall")
 local histogram = require("core.histogram")
+local json = require("lib.json")
+local macaddress = require("lib.macaddress")
 local usage = require("program.top.README_inc")
 
 local long_opts = {
-   help = "h"
+   help = "h", counters = "c", yang = "y"
 }
 
 function clearterm () io.write('\027[2J') end
 
 function run (args)
    local opt = {}
+   local object = nil
+   local yang = false
    function opt.h (arg) print(usage) main.exit(1) end
-   args = lib.dogetopt(args, opt, "h", long_opts)
+   function opt.c (arg) object = arg              end
+   function opt.y ()    yang = true               end
+   args = lib.dogetopt(args, opt, "hc:y", long_opts)
 
    if #args > 1 then print(usage) main.exit(1) end
-   local target_pid = args[1]
+   local target_pid = select_snabb_instance(args[1])
 
-   local instance_tree = "//"..(select_snabb_instance(target_pid))
+   if     object then list_counters(target_pid, object)
+   elseif yang   then dump_yang(target_pid)
+   else               top(target_pid) end
+   ordered_exit(0)
+end
+
+function select_snabb_instance (pid)
+   local instances = shm.children("//")
+   if pid then
+      -- Try to use given pid
+      for _, instance in ipairs(instances) do
+         if instance == pid then return pid end
+      end
+      print("No such Snabb instance: "..pid)
+   elseif #instances == 2 then
+      -- Two means one is us, so we pick the other.
+      local own_pid = tostring(S.getpid())
+      if instances[1] == own_pid then return instances[2]
+      else                            return instances[1] end
+   elseif #instances == 1 then print("No Snabb instance found.")
+   else print("Multple Snabb instances found. Select one.") end
+   ordered_exit(1)
+end
```

Comment:
Is [DWIM](https://en.wikipedia.org/wiki/DWIM) a good approach to deciding which Snabb process to monitor?

Alternative would be to invent a simple and unambiguous way to refer to a particular process e.g. by name.

Related issue: `lugano-1` currently has [1,657](https://gist.github.com/lukego/ec282a9fde4f2835317c536131586e4d) stale process directories in `/var/run/snabb`. Do people really run `snabb gc` at appropriate intervals in practice? This is relevant to me now because I would like to create relatively large timeline logs in shared memory but I don't want this to linger around chewing up scarce space on a ramdisk folder i.e. continuing to consume memory after termination.